tree_migration_milestone1.js

```diff
@@ -0,0 +1,125 @@
+require("dotenv").config();
+const { knex } = require("./database/knex");
+
+var ProgressBar = require("progress");
+
+async function migrate() {
+  try {
+    const base_query_string =
+      "from public.trees t left join field_data.raw_capture r on t.uuid = r.id::text where t.approved=true and r.id is null";
+
+    const count = await knex.select(knex.raw(`count(*) ${base_query_string}`));
+    const query_stream = knex.raw(`select t.* ${base_query_string}`).stream();
+
+    var bar = new ProgressBar("Migrating [:bar] :percent :etas", {
+      width: 20,
+      total: +count[0].count,
+    });
+
+    const attributes_formatter = (attributes) => {
+      if (attributes.length <= 0) return null;
+      const attributes_in_json_format = { entries: [] };
+      for (const { key, value } of attributes) {
```

Comment:
What about this loop as the following, it avoids instantiating new objects.

```
for(let entry of attributes) {
   attributes_in_json_format.entries.push(entry);
}
```

Not that it is terribly important since it is a script, but was wondering the implication of I am suggesting here versus what you have.